docs/modules/user-guide/partials/proc_defining-variables.adoc

```diff
@@ -0,0 +1,40 @@
+[id="proc_defining-variables_{context}"]
+= Defining variables
+
+[role="_abstract"]
+The `variables` object is a map of variable name-value pairs that are used for string replacement in the devfile. Variables are referenced using the syntax `+{{variable-name}}+` to insert the corresponding value in string fields in the devfile. 
+
+Variables can be defined at the top level of the devfile or in the `parent` object. String replacement with variables cannot be used for:
```

Comment:
Would it help to rewrite this for the reader?  For example:

`As a <your role: developer, administrator>, you can define which variables devfiles replaces with strings. <the benefit of doing this>. etc...`

Reply:
Keeping the passive voice here obscures who or what performs the action and whether the object is static or mutable. Or assumes that the reader already knows this information.

Reply:
Most of the devfiles content has been shoehorned into procedures, when it should be reference content - we are documenting a syntax/schema, similar to an API.

Having stand-alone procedures telling a user how to add one line to a file, for example, saying "version: 2.2.0", is not a good use of procedures.

With reference content, I feel passive voice can be more appropriate at times. 

Separately, the new docs platform for devfiles does not (yet) support modular content - so all those redundant assemblies that only contain a single include to a module will have to be replace with files containing the content itself.